app/js/deleteCategory.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * deletes category from database
+ * @param {number} category id
+ * @returns {Promise<*>} response from api
+ */
+async function deleteCategory(id: number) {
```

Comment:
Missing a return type hint. If it's returning a promise - you can type it `Promise<shapeOfResponse>`